src/scenes/UserGalleryPage/QRCode.tsx

```diff
@@ -0,0 +1,207 @@
+import colors from 'components/core/colors';
+import styled from 'styled-components';
+import { useCallback, useState, useRef } from 'react';
+import Spacer from 'components/core/Spacer/Spacer';
+// import QRCodeStyling from 'qr-code-styling';
+
+function CloseIcon() {
+  return (
+    <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+      <path d="M12.6667 3.33334L3.33333 12.6667" stroke="black" stroke-miterlimit="10" />
+      <path d="M3.33333 3.33334L12.6667 12.6667" stroke="black" stroke-miterlimit="10" />
+    </svg>
+  );
+}
+
+function QRIcon() {
+  return (
+    <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+      <path d="M22 6V2H18" stroke="black" stroke-miterlimit="10" />
+      <path d="M2 18V22H6" stroke="black" stroke-miterlimit="10" />
+      <path d="M18 22H22V18" stroke="black" stroke-miterlimit="10" />
+      <path d="M6 2H2V6" stroke="black" stroke-miterlimit="10" />
+      <path d="M10.5 5.5H5.5V10.5H10.5V5.5Z" stroke="black" stroke-miterlimit="10" />
+      <path d="M18.5 5.5H13.5V10.5H18.5V5.5Z" stroke="black" stroke-miterlimit="10" />
+      <path d="M13.5 19V16V13" stroke="black" stroke-miterlimit="10" />
+      <path d="M18.5 16V18.5H16" stroke="black" stroke-miterlimit="10" />
+      <path d="M16 13.5H19" stroke="black" stroke-miterlimit="10" />
+      <path d="M10.5 13.5H5.5V18.5H10.5V13.5Z" stroke="black" stroke-miterlimit="10" />
+      <path d="M7.5 8H8.5" stroke="black" stroke-miterlimit="10" />
+      <path d="M15.5 8H16.5" stroke="black" stroke-miterlimit="10" />
+      <path d="M7.5 16H8.5" stroke="black" stroke-miterlimit="10" />
+      <path d="M16.5 16H13.5" stroke="black" stroke-miterlimit="10" />
+    </svg>
+  );
+}
+
+export default function QRCode({ username }: { username: string }) {
+  const ref = useRef<HTMLDivElement>(null);
+  const [showQRCode, setShowQRCode] = useState(false);
+  const [qrCodeExists, setQRCodeExists] = useState(false);
+
+  const handleClick = useCallback(() => {
+    setShowQRCode(true);
+
+    // Dynamically import qr-code-styling on the client-side, not compatible with regular Next imports
+    // https://github.com/kozakdenys/qr-code-styling/issues/38
+    if (typeof window !== 'undefined') {
+      if (qrCodeExists) return; // Prevent duplication
+      const QRCodeStyling = require('qr-code-styling');
+
+      const qrCode = new QRCodeStyling({
+        width: 636, // 4 * 159px, scaled down using CSS later
+        height: 636, // 4 * 159px, scaled down using CSS later
```

Comment:
I ran into some resolution issues with the initial height and width. Here, we render the canvas element with a width and height of 636px (4x the initial) and then we scale it down to its proper width and height (159px) using CSS. This makes the QR appear high res

Reply:
nicee thanks for the comment - I think it adds enough context that it would be useful to add this comment (or a summary) to the file

Reply:
can definitely see myself or someone new in the future wondering why its 4x

Reply:
+1 thank you for the comments!